### PR TITLE
Create spike for Cypress E2E testing

### PR DESCRIPTION
**Pull request by patcon**
Resolves https://github.com/GCTC-NTGC/gc-digital-talent/issues/1762

- [x] install cypress in `frontend/`
- [x] add scaffold files
- [x] add simple starter tests
    - footer
    - static pages (ToS and privacy)
    - language selection page
- [x] increase quality of generated video artefacts so that debugging is easier
- [x] fixed outdated editorconfig file
- [x] wrote docs for cypress testing and custom commands
- [x] add github action workflow
  - [x] docker layer caching preserved between workflows, so 3s rather than 2min
    - layers cached between workflows: docker smoke tests, and past runs of this
  - [x] cancel other versions of workflow that are mid-run (concurrency settings), to not waste build minutes
  - [x] save screenshots as artifacts on failure
  - [x] save video screencasts of tests as artifacts (see summary page to download, great for debugging)
- [x] add reporter to print Cypress logs for failed tests (that would normally show in Cypress console) to CI terminal logs
  - https://github.com/archfz/cypress-terminal-report
- [x] ~~enable junit xml reporter for tests and report via github actions~~ `wont-do`
  - https://github.com/EnricoMi/publish-unit-test-result-action
  - https://docs.cypress.io/guides/tooling/reporters#Reporter-Options
  - https://github.com/bhovhannes/junit-report-merger
- [x] enable multi-reporter to still get output into terminal logs
  - https://github.com/you54f/cypress-multi-reporters

### CODE REVIEW

**Comment by patcon** [comment]
Yay! It works!

It takes quite some time to run `setup.sh`, but the docker build went from 1min to 3s with the layer caching, so that's a good sign that this will be _real_ quick once we have the app actually building in docker :)

Screen Shot 2022-04-26 at 8 09 00 PM

cc: @petertgiles

**Comment by patcon** [comment]
Going to back out of the reporting goal, as it's a little fraught.

Examples:
- dorny's test-reporter using java-junit reporter: https://github.com/patcon/gc-digital-talent/runs/6186727684
- using another reporter action
  - https://github.com/patcon/gc-digital-talent/pull/288/checks?check_run_id=6186621165
  - https://github.com/patcon/gc-digital-talent/pull/270#issuecomment-1110444893

both reporters took awhile to debug because the check results are showing up under "Bundlewatch" check for some reason, which is perhaps due to "Bundlewatch" action doing something non-standard.

Also, there's some confusing work to make GitHub check reports work for forks:
https://github.com/EnricoMi/publish-unit-test-result-action/tree/v1.20#support-fork-repositories-and-dependabot-branches

Also, cypress has an issue with mocha reporters can't have their output locations configured right now:
https://github.com/cypress-io/cypress/issues/18014

tl;dr - too many rabbit holes here!

**Comment by patcon** [comment]
@petertgiles the interface won't see to let me, but I'm undesignating you as reviewer! (I am too quick to tag you, and it's a bad habit 🙂 )

@esizer @yonikid15 you seem like you'd get a kick out of e2e tests. Care to review when you get a sec? :)

**Review by esizer — CHANGES_REQUESTED** [review]
Looks great! Just noticed a small change to the footer test descriptions

File: frontend/cypress/integration/talentsearch/footer.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+describe('Footer', () => {
+  context('English page', () => {
+    beforeEach(() => cy.visit('/en'))
+    it('links to Privacy Policy', () => {
```
[the remark this answers is not in the document]
Comment (patcon) [reply frontend/cypress/integration/talentsearch/footer.spec.js:4]: Good catch! Fixed.

**Review by esizer — APPROVED** [review]
Sweet! 🍭
